utils/helpers.ts

```diff
@@ -151,3 +151,62 @@ export function checkSearchTerm(searchTerm: string, plant: Plant) {
   // Check if plant_name contains searchTerm
   return plant.plant_name.toLowerCase().includes(searchTerm.toLowerCase());
 }
+
+export function checkSunlight(
+  sunlightFilterValue: DropdownOption[],
+  plant: Plant,
+) {
+  // Automatically returns true if no selected sunlight
+  if (sunlightFilterValue.length === 0) {
+    return true;
+  }
+
+  const sunlightToHours = new Map<string, [number, number]>([
+    ['SHADE', [0, 2]],
+    ['PARTIAL_SHADE', [2, 4]],
+    ['PARTIAL_SUN', [4, 6]],
+    ['FULL', [6, Number.MAX_VALUE]],
+  ]);
+
+  // For each sunlight selected, check if plant's min_hours and max_hours are
+  // within that enum's range
+  const sunlightBoolean: boolean[] = [];
+  for (const sunlight of sunlightFilterValue) {
+    const [minHours, maxHours] = sunlightToHours.get(sunlight.value)!;
+    // if max_hours is null then plant can only receive min_hours of sunlight
+    if (plant.sunlight_max_hours === null) {
+      sunlightBoolean.push(
```

Comment:
nit: instead of creating an array `sunlightBoolean` and pushing to it for each sunlight value, we can just directly return true if the plant matches any of the conditions

`if (plant.sunlight_min_hours >= minHours && plant.sunlight_min_hours <= maxHours) return true;`

and `return false` at the end. 

I think the current implementation is fine, but I'm just curious if there was a reason (i.e. if there was any benefit) over the alternate implementation suggested here.